### PR TITLE
SRVKS-261: Removing Knative Serving remains

### PR DESCRIPTION
**Pull request by nebelsprachen**
_Labels_:
* enteprise-4.1
* enteprise-4.2

------------------------------------------------------------------------------------
* Deletion of Knative Serving

### CODE REVIEW

File: modules/serverless-deleting-knative-serving-crds.adoc [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// serverless/installing-openshift-serverless.adoc
+
+[id="deleting-knative-serving-crds_{context}"]
+= Deleting Knative Serving CRDs
+
+After uninstalling the {ServerlessOperatorName}, some Knative Serving components may remain. Use this procedure to completely uninstall Knative Serving API services and CRDs.
```
Comment (markusthoemmes) [inline modules/serverless-deleting-knative-serving-crds.adoc]:
```suggestion
After uninstalling the {ServerlessOperatorName}, the CRDs and API Services of the operator itself remain on the cluster. Use this procedure to remove those.
```

File: modules/serverless-deleting-knative-serving-crds.adoc [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// serverless/installing-openshift-serverless.adoc
+
+[id="deleting-knative-serving-crds_{context}"]
+= Deleting Knative Serving CRDs
```
Comment (markusthoemmes) [inline modules/serverless-deleting-knative-serving-crds.adoc]:
```suggestion
= Deleting Knative Serving CRDs from the operator
```

File: modules/serverless-deleting-knative-serving-crds.adoc [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+After uninstalling the {ServerlessOperatorName}, some Knative Serving components may remain. Use this procedure to completely uninstall Knative Serving API services and CRDs.
+
+.Prerequisite
+*  You have deleted the `knative-serving` project and removed the {ServerlessOperatorName}.
```
Comment (markusthoemmes) [inline modules/serverless-deleting-knative-serving-crds.adoc]:
```suggestion
*  You have uninstalled Knative Serving and removed the {ServerlessOperatorName} per above instructions.
```

**Review by markusthoemmes — COMMENTED** [review]
I think above the `Deleting the OpenShift Serverless Operator` section there should be an `Uninstalling Knative Serving` section, which simply states that one shall remove the `KnativeServing` and namespace that was created with:

```
oc delete knativeserving knative-serving -n knative-serving
```

Then wait until it's all gone to avoid deadlocks, then do:

```
oc delete namespace knative-serving
```

---

Thanks for the writeup 🙂

File: modules/serverless-uninstalling-knative-serving.adoc [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+. To remove Knative Serving, use the following command:
++
+----
+oc delete knativeserving knative-serving -n knative-serving
+----
+
+. After the command has completed, use the following command to delete the `knative-serving` namespace:
++
+----
+ delete namespace knative-serving
```
Comment (markusthoemmes) [inline modules/serverless-uninstalling-knative-serving.adoc]:
```suggestion
oc delete namespace knative-serving
```

File: modules/serverless-uninstalling-knative-serving.adoc [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+.Procedure
+
+. To remove Knative Serving, use the following command:
++
+----
+oc delete knativeserving knative-serving -n knative-serving
+----
+
+. After the command has completed, use the following command to delete the `knative-serving` namespace:
```
Comment (markusthoemmes) [inline modules/serverless-uninstalling-knative-serving.adoc]:
```suggestion
. After the command has completed and all pods have vanished from the `knative-serving` namespace, remove the namespace with the following command:
```

**Review by markusthoemmes — APPROVED** [review]
I like it, thank you!

LGTM

File: modules/serverless-deleting-knative-serving-crds.adoc [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+//
+// serverless/installing-openshift-serverless.adoc
+
+[id="deleting-knative-serving-crds_{context}"]
+= Deleting Knative Serving CRDs from the Operator
+
+After uninstalling the {ServerlessOperatorName}, the Operator CRDs and API services remain on the cluster. Use this procedure to completely uninstall the remaining components. 
+
+.Prerequisite
+*  You have uninstalled Knative Serving and removed the {ServerlessOperatorName} using the procedure above.
```
Comment (huffmanca) [inline modules/serverless-deleting-knative-serving-crds.adoc]: Per the IBM style guide, don't use `above`. Use `previously` or `preceding` to indicate position.

**Review by huffmanca — COMMENTED** [review]
Largely looks good, just one minor nitpick regarding the word `above`.

**Pull request merged**